src/imgix-core-js.js

```diff
@@ -186,23 +192,29 @@
     ImgixClient.prototype._buildDPRSrcSet = function(path, params, options) {
         var srcset = '';
         var targetRatios = [1, 2, 3, 4, 5];
-        var currentRatio;
         var disableVariableQuality = options.disableVariableQuality || false;
-        var quality = params.q;
+
+        var key;
+        var queryParams = {};
+        for (key in params) {
+          queryParams[key] = params[key];
+        }
```

Comment:
Nit because I can. So in ES2015 this kind of behaviour is provided by the function "Object.assign". We can't use it (yet) because it's not supported in IE 11, but maybe this could be written as

```suggestion
var queryParams = assign({}, params);
```

and then have a helper method somewhere else that is:

```js
var assign = function assign() {
  var target = arguments[0];
  var rest = arguments.slice(1);

  rest.map(function (obj) { 
    for (key in obj) {
      target[key] = obj[key];
    }
  }); 

  return target
} 
```

Then the code in the for loop can be better represented as (since e.g. the assigning to `queryParams.dps` is a bit of a smell):

```js
for (var i = 0; i < targetRatios.length; i++) {
  var currentRatio = targetRatios[i];

  if (!disableVariableQuality) {
    var q = quality || DPR_QUALITIES[currentRatio];
  }

  srcset +=
    this.buildURL(path, assign({}, queryParams, { dpr: currentRatio, q: q })) + " " + currentRatio + "x,\n";
}
```

Reply:
Yeah, I would have reached for `Object.assign` if it was an option. I would also agree that (most) assignment is a bit of a smell. Although there's a cleaner way of doing the above, I figured we could save that for a future refactor?

It also feels like `sequence.map(forLoop)` may also be a bit of a smell, but then again I'm not sure if this is something that's done frequently in JS. Lmk! ha

Reply:
Yeah you're right `sequence.map(forLoop)` is a bit of a smell, maybe you could use a `.reduce` instead. 

```js
var assign = function assign() {
  return (arguments || []).reduce(function (p, v, i) {
    if (i === 0) { return v };
	Object.keys(v).map(function (key) { p[key] = v[key });
    return p;
  }, {});
} 
```

We could also use the official polyfill from MDN:
```js
if (typeof Object.assign !== 'function') {
  // Must be writable: true, enumerable: false, configurable: true
  Object.defineProperty(Object, "assign", {
    value: function assign(target, varArgs) { // .length of function is 2
      'use strict';
      if (target === null || target === undefined) {
        throw new TypeError('Cannot convert undefined or null to object');
      }

      var to = Object(target);

      for (var index = 1; index < arguments.length; index++) {
        var nextSource = arguments[index];

        if (nextSource !== null && nextSource !== undefined) { 
          for (var nextKey in nextSource) {
            // Avoid bugs when hasOwnProperty is shadowed
            if (Object.prototype.hasOwnProperty.call(nextSource, nextKey)) {
              to[nextKey] = nextSource[nextKey];
            }
          }
        }
      }
      return to;
    },
    writable: true,
    configurable: true
  });
}
```

Reply:
Small follow up note. I think, with the way the function interfaces currently work together , assigning to `queryParams.dps` is necessary because `this.buildURL(path, queryParams)` (ultimately `.__buildParams`) uses the key-value pairs in the `queryParams` object to build the query parameter string.

I think there's a more functional approach that eliminates assignment in a few places, but I figured we'd leave that for a future refactor.